hypothesis-python/RELEASE.rst

```diff
@@ -0,0 +1,3 @@
+RELEASE_TYPE: patch
+
+Add more type hints to our internals (:issue:`3074`).
```

Comment:
Let's note the changes to `st.nothing()` type annotations?  That's a nice user-visible feature!